lib/ddtrace/contrib/cucumber/patcher.rb

```diff
@@ -0,0 +1,54 @@
+require 'ddtrace/contrib/patcher'
+require 'ddtrace/ext/integration'
+require 'ddtrace/ext/ci'
+require 'ddtrace/ext/net'
+require 'ddtrace/contrib/analytics'
+
+module Datadog
+  module Contrib
+    module Cucumber
+      # Patcher enables patching of 'cucumber' module.
+      module Patcher
+        include Contrib::Patcher
+
+        module_function
+
+        def target_version
+          Integration.version
+        end
+
+        def patch
+          require 'ddtrace/pin'
+
+          Datadog::Pin.new(
+            Datadog.configuration[:cucumber][:service_name],
+            app: Datadog::Contrib::Cucumber::Ext::APP,
+            app_type: Datadog::Ext::AppTypes::TEST,
+            tags: Datadog::Ext::CI.tags(ENV).merge(Datadog.configuration.tags),
+            tracer: -> { Datadog.configuration[:cucumber][:tracer] }
+          ).onto(::Cucumber)
+
+          patch_cucumber_runtime
+        end
+
+        def patch_cucumber_runtime
+          require 'ddtrace/contrib/cucumber/formatter'
+
+          ::Cucumber::Runtime.class_eval do
```

Comment:
We try not to use `class_eval` when possible, it's brittle and can cause conflicts with other gems. Using a combination of `include` and `prepend` here is the suggested approach `httprb` can be a guide for this

https://github.com/DataDog/dd-trace-rb/blob/4937c97e935015f7754fb971a5bdad573f1a08cc/lib/ddtrace/contrib/httprb/instrumentation.rb#L14-L16

https://github.com/DataDog/dd-trace-rb/blob/4937c97e935015f7754fb971a5bdad573f1a08cc/lib/ddtrace/contrib/httprb/patcher.rb#L26